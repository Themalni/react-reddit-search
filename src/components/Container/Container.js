import styled from 'styled-components';

const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
`
const LowerContainer = styled(Container)`
  margin-top: 2em;
`
const MasonryContainer = styled.div`
  width: 100%;
  margin-top: 2em;
  display: flex;
  flex-flow: column;
  align-content: center;

  ${(props) => props.theme.media.tablet} {
    width: 90%;
    flex-flow: column wrap;
    max-height: 4000px;
  }
  ${(props) => props.theme.media.desktop} {
    width: 75%;
  }
`

export { Container, LowerContainer, MasonryContainer }