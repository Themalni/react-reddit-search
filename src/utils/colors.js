export const colors = {
  white: '#fff',
  graphite: '#666',
  grey: '#8a96a3',
  yellowOrange: '#ff9933',
  orange: '#ed5f55',
  orangeDark: '#d8574e',
  blue: '#3a81b7',
  blueLight: '#64a0ce'
};

// d02216