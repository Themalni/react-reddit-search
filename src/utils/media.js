export const media = {
  phone: '@media (min-width: 320px)',
  tablet: '@media (min-width: 728px)',
  laptop: '@media (min-width: 1024px)',
  desktop: '@media (min-width: 1280px)'
}