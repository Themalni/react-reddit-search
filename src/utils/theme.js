import { colors } from './colors';

export const theme = {
  colors,
  font: {
    thin: 300,
    regular: 400,
    bold: 700,
  },
  media: {
    phone: '@media (min-width: 320px)',
    tablet: '@media (min-width: 728px)',
    laptop: '@media (min-width: 1024px)',
    desktop: '@media (min-width: 1280px)'
  }
}