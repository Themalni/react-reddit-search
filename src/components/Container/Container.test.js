import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components'
import { theme } from '../../utils/theme';
import { Container, LowerContainer, MasonryContainer } from './Container';
import { ThemeProvider } from 'styled-components';

describe('Container', () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Container />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot();
  })
})

describe('LowerContainer', () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <LowerContainer />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot();
  })
})

describe('MasonryContainer', () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MasonryContainer />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot();
  })
})