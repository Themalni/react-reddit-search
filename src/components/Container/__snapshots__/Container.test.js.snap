// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
/>
`;

exports[`LowerContainer renders correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 2em;
}

<div
  class="c0"
/>
`;

exports[`MasonryContainer renders correctly 1`] = `
.c0 {
  width: 100%;
  margin-top: 2em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

@media (min-width:728px) {
  .c0 {
    width: 90%;
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    max-height: 4000px;
  }
}

@media (min-width:1280px) {
  .c0 {
    width: 75%;
  }
}

<div
  class="c0"
/>
`;
